src/wp-includes/script-loader.php

```diff
@@ -1843,6 +1843,34 @@ function wp_just_in_time_script_localization() {
 	);
 }
 
+/**
+ * Prints a loader script if there is text/template registered script.
+ *
+ * @since 6.3.0
+ *
+ * When added to the DOM, this script converts any text/template scripts
+ * associated with a handle into type/javascript, and executes them.
+ */
+function wp_print_delayed_inline_script_loader() {
+	$wp_scripts = wp_scripts();
+
+	if ( $wp_scripts->has_delayed_inline_script() ) {
+		$output    = <<<JS
+function wpLoadAfterScripts( handle ) {
+	const scripts = document.querySelectorAll(`[type="text/template"][data-wp-executes-after="\${handle}"]`);
+	scripts.forEach( (script) => {
+		script.setAttribute("type","text/javascript");
+		eval(script.textContent);
```

Comment:
Has the presence of `eval()` been considered for how it could raise red flags for security vulnerability scanners? What about for sites that are using CSP? If the [`unsafe-eval`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src#unsafe_eval_expressions) source expression is not explicitly passed, then `eval()` would be blocked here. For that matter, there's also [Core-32067](https://core.trac.wordpress.org/ticket/32067) which proposes the removal of all inline scripts, but that is even more extreme. @adamsilverstein I seem to recall you being involved with CSP years ago in an advisory capacity?

Reply:
Why is `eval` needed here in the first place? Couldn't we just replace the `script[type="text/template"]` tag with the actual `script` tag in the DOM and it will simply be executed?

Reply:
Indeed. The following function seems to work just as well, and it avoids the use of `eval()`, while also being ES5-compatible:

```js
function wpLoadAfterScripts(handle) {
  var scripts, newScript, i, len;
  scripts = document.querySelectorAll(
    '[type="text/template"][data-wp-executes-after="' + handle + '"]'
  );
  for (i = 0, len = scripts.length; i < len; i++) {
    newScript = scripts[i].cloneNode(true);
    newScript.type = "text/javascript";
    scripts[i].parentNode.replaceChild(newScript, scripts[i]);
  }
}
```

Reply:
Demo: https://codepen.io/westonruter/pen/JjmJNVX?editors=1000

Reply:
@westonruter I tested this as a replacement and it ran perfectly fine. I will commit this shortly, I just need to update the associated tests.

Reply:
@westonruter this has been updated, please see https://github.com/10up/wordpress-develop/pull/54/files#r1179634959

Reply:
@westonruter and @joemcgill I am going to resolve this comment thread in light of my previous reply.